### PR TITLE
Trim smtp

### PR DESCRIPTION
**Pull request by mbaechler**
you can review if you want, but I know that there's ticket number missing.

### CODE REVIEW

**Comment by chibenwa** [comment]
I do always really like your PRs ^^

If interesting, we can always create ticket while merging don't worry.

File: server/container/guice/guice-common/src/main/java/org/apache/james/modules/MissingArgumentException.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.apache.james.modules;
```
Comment (chibenwa) [inline server/container/guice/guice-common/src/main/java/org/apache/james/modules/MissingArgumentException.java]: License

File: server/container/guice/jmx/pom.xml [hunk 2]
```diff
@@ -0,0 +1,212 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chibenwa) [inline server/container/guice/jmx/pom.xml]: Licence

**Comment by chibenwa** [comment]
Btw why trim SMTP ?

**Comment by mbaechler** [comment]
> Btw why trim SMTP ?

Because it's one of the 3 use cases we decided to have in james 3 : full server with distributed backend (C* + ES), full server in single node (JPA + lucene) and smtp only for mailet pipeline.

And I don't like pulling tons of useless projects for that use case.

More generally, we need to clean up our dependencies and starting with smtp is a good test case.

**Comment by mbaechler** [comment]
> I would propose you to create a guice-mailbox-default dependancy where you can put these two modules. 

There's already a guice-mailbox. I think that default impl could be put there for now.

File: server/container/guice/jmx/pom.xml (hunk 2 above)
Comment (chibenwa) [inline server/container/guice/jmx/pom.xml]: Still no license

**Comment by chibenwa** [comment]
Merged with added licenses and created Apache ticket

**Pull request closed without merging**